linera-views/src/dynamo_db.rs

```diff
@@ -324,61 +320,68 @@ pub struct DynamoDbStoreInternal {
 pub struct DynamoDbStoreConfig {
     /// The AWS configuration
     pub config: Config,
-    /// The table_name used
-    pub table_name: TableName,
     /// The common configuration of the key value store
     pub common_config: CommonStoreConfig,
 }
 
-impl DynamoDbStoreInternal {
-    fn build_delete_transact(
-        &self,
-        key: Vec<u8>,
-    ) -> Result<TransactWriteItem, DynamoDbContextError> {
-        ensure!(!key.is_empty(), DynamoDbContextError::ZeroLengthKey);
-        ensure!(key.len() <= MAX_KEY_SIZE, DynamoDbContextError::KeyTooLong);
-        let request = Delete::builder()
-            .table_name(&self.table.0)
-            .set_key(Some(build_key(key)))
-            .build()?;
-        Ok(TransactWriteItem::builder().delete(request).build())
+#[async_trait]
+impl AdminKeyValueStore<DynamoDbContextError> for DynamoDbStoreInternal {
+    type Config = DynamoDbStoreConfig;
+
+    async fn connect(config: &Self::Config, namespace: &str) -> Result<Self, DynamoDbContextError> {
+        let client = Client::from_conf(config.config.clone());
+        Self::check_namespace(namespace)?;
+        let semaphore = config
+            .common_config
+            .max_concurrent_queries
+            .map(|n| Arc::new(Semaphore::new(n)));
+        let max_stream_queries = config.common_config.max_stream_queries;
+        let namespace = namespace.to_string();
+        Ok(Self {
+            client,
+            namespace,
+            semaphore,
+            max_stream_queries,
+        })
     }
 
-    fn build_put_transact(
-        &self,
-        key: Vec<u8>,
-        value: Vec<u8>,
-    ) -> Result<TransactWriteItem, DynamoDbContextError> {
-        ensure!(!key.is_empty(), DynamoDbContextError::ZeroLengthKey);
-        ensure!(key.len() <= MAX_KEY_SIZE, DynamoDbContextError::KeyTooLong);
-        ensure!(
-            value.len() <= RAW_MAX_VALUE_SIZE,
-            DynamoDbContextError::ValueLengthTooLarge
-        );
-        let request = Put::builder()
-            .table_name(&self.table.0)
-            .set_item(Some(build_key_value(key, value)))
-            .build()?;
-        Ok(TransactWriteItem::builder().put(request).build())
+    async fn list_all(config: &Self::Config) -> Result<Vec<String>, DynamoDbContextError> {
+        let client = Client::from_conf(config.config.clone());
+        let mut namespaces = Vec::new();
+        let mut start_table = None;
+        loop {
+            let response = client
+                .list_tables()
+                .set_exclusive_start_table_name(start_table)
+                .send()
+                .await?;
+            if let Some(namespaces_blk) = response.table_names {
+                namespaces.extend(namespaces_blk);
+            }
+            if response.last_evaluated_table_name.is_none() {
```

Comment:
(I guess this will also go away once we use `query_iter`?)

Reply:
Yes. If we move to `query_iter`.